blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/BranchContainerImpl.java

```diff
@@ -145,15 +145,20 @@ public BluePipeline get(String name) {
     @Override
     @SuppressWarnings("unchecked")
     public Iterator<BluePipeline> iterator(int start, int limit) {
-        List<BluePipeline> branches = new ArrayList<>();
-        Collection<Job> jobs = pipeline.mbp.getAllJobs();
-
-        jobs = ContainerFilter.filter(jobs, start, limit);
-        for(Job j: jobs){
-            branches.add(new BranchImpl(j, getLink()));
-        }
-
-        return Ordering.from(BRANCH_COMPARITOR).sortedCopy(branches).iterator();
+        final Link link = getLink();
+        // Filter will decide if the requester wants branches or pull requests
+        Collection allJobsMatchinFilter = ContainerFilter.filter(pipeline.mbp.getAllJobs());
```

Comment:
This will regress fix for JENKINS-44867, as filter uses Action to determine PRs, can be very expensive. `ContainerFilter.filter(jobs, start, limit)` counts up to limit items based on filter then they get sorted later. Maybe client could send limit=500 or so to limit the damage?